File: Data Structures & Algorithms/sort-colors/submission-0.java
class Solution {
    public void sortColors(int[] nums) {
        int low = 0;
        int mid = 0;
        int high = nums.length - 1;

        while(mid<=high){
            if(nums[mid] == 0){
                int temp = nums[low];
                nums[low] = nums[mid];
                nums[mid] = temp;
                low = low+1;
                mid = mid+1;
            }else if(nums[mid] == 1){
                mid=mid+1;
            }else if(nums[mid] == 2){
                int temp = nums[high];
                nums[high] = nums[mid];
                nums[mid] = temp;
                high = high-1;
            }
        }
    }
}

File: Data Structures & Algorithms/find-target-in-rotated-sorted-array/submission-0.py
class Solution:
    def search(self, nums: List[int], target: int) -> int:
        n = len(nums)
        l,r = 0, len(nums) - 1
        while l<r:
            k = (l+r) // 2
        
            if nums[k] > nums[r]:
                l = k + 1
            else:
                r = k
        
        pivot =  l


        if target >=nums[pivot] and target <= nums[n-1]:
            l,r = pivot, n-1
        else:
            l,r = 0, pivot - 1


        while l<=r:
            mid = (l+r)//2
            if nums[mid] == target:
                return mid
            elif target < nums[mid]:
                r = mid-1
            else:
                l = mid+1

        
        return -1